c_binding/libses.c

```diff
@@ -723,13 +695,100 @@ int _ses_dev_slot_ctrl(char *err_msg, const char *tp_sas_addr,
     }
 
  out:
-    if (fd != -1)
+    if (fd >= 0)
         close(fd);
+    return rc;
+}
+
+static int _ses_info_get_by_sas_addr(char *err_msg, const char *tp_sas_addr,
+                                      uint8_t *cfg_data, uint8_t *status_data,
+                                      uint8_t *add_st_data, int *fd,
+                                      int16_t *element_index)
+{
+    int rc = LSM_ERR_OK;
+    char  **sg_paths = NULL;
+    uint32_t sg_count = 0;
+    uint32_t i = 0;
+    bool found = false;
+
+    assert(tp_sas_addr != NULL);
+    assert(cfg_data != NULL);
+    assert(status_data != NULL);
+    assert(add_st_data != NULL);
+    assert(element_index != NULL);
+    assert(fd != NULL);
+
+    _good(_ses_sg_paths_get(err_msg, &sg_paths, &sg_count), rc, out);
+
+    for (i = 0; i < sg_count; ++i) {
+        _good(_sg_io_open_rw(err_msg, sg_paths[i], fd), rc, out);
+        _good(_sg_io_recv_diag(err_msg, *fd, _T10_SES_CFG_PG_CODE, cfg_data),
```

Comment:
On line 724 we open the fd, if we fail anywhere between there and 742, by jumping to the goto I believe we leak the fd?